src/python/grpcio/commands.py

```diff
@@ -211,6 +212,20 @@ class BuildExt(build_ext.build_ext):
     }
     LINK_OPTIONS = {}
 
+    def get_ext_filename(self, ext_name):
+      # since python3.5, python extensions' shared libraries use a suffix that corresponds to the value
+      # of sysconfig.get_config_var('EXT_SUFFIX') and contains info about the architecture the library targets.
+      # E.g. on x64 linux the suffix is ".cpython-XYZ-x86_64-linux-gnu.so"
+      # When crosscompiling python wheels, we need to be able to override this suffix
+      # so that the resulting file name matches the target architecture and we end up with a well-formed
+      # wheel.
+      filename = build_ext.build_ext.get_ext_filename(self, ext_name)
+      orig_ext_suffix = sysconfig.get_config_var('EXT_SUFFIX')
+      new_ext_suffix = os.getenv('GRPC_PYTHON_OVERRIDE_EXT_SUFFIX')
+      if new_ext_suffix and filename.endswith(orig_ext_suffix):
+        filename = filename[:-len(orig_ext_suffix)] + new_ext_suffix
```

Comment:
Nit: `filename = filename.replace(orig_ext_suffix, new_ext_suffix, 1)` might be more readable?

Reply:
what I have is a bit more complicated, but it's more robust. I only want to replace a substring if it's a suffix. Theoretically, you can have the name of the extension that contains the same string? (but yeah, it's quite unlikely).
I might just use replace.

Reply:
That's true. Leaving it up to you.